syft/frameworks/torch/tensors/interpreters/replicated_shared.py

```diff
@@ -306,3 +309,7 @@ def __str__(self):
             for v in self.child.values():
                 out += "\n\t-> " + str(v)
         return out
+
+
+### Register the tensor with hook_args.py ###
```

Comment:
what is that?

Reply:
its been added so that the  _hook_method_args of overload_method can know Put back SyftTensor on the tensors found in the response. if this was not used the output of add or sub would have (Wrapper)>[ReplicatedSharingTensor] instead of (Wrapper)>FixedPrecisionTensor>[RaplicatedSharingTensor] after the operation.

Reply:
this is not the right place, it breaks the level of abstraction torch hook is a lower-level abstraction so RST should not know it exists.

Reply:
@gmuraru can you help me this issue?

Reply:
Yep - this is needed because when we add the wrapper we check what tensors exists - if we do not have this here it would not know how to create the ```(Wrapper)>RST```

Reply:
A better approach to this (for all syft framework) it should be to add something "invasive" like:
```
1. get all classes that inherit from AbstractTensor
2. call this ```default_register_tensor``` on all of them
```
But I think another issue might be in order to do this -- doing one now.

Reply:
we are doing this in other issue, right? @gmuraru @abogaziah